### DISCUSSION TITLE
Updating the state within a custom node?

### DISCUSSION THREAD

**Discussion by NahanTu** [xyflow/xyflow#1451] [Q&A]
I am trying to create a new node within a custom node (click on button on custom node -> shall create a new node and update the state). But when I try to use
`useStoreActions((action) => action.setElements)`
I get a TypeError: propElements.reduce is not a function.
Can anybody give me an example of how to properly use the action.setElements in order to add a new node into the state please?

**Comment by FredrikAugust** [comment]
I use my own Redux store on top of Flow for managing the elements, and pass the state of this into the React Flow element. With this I'm able to `useDispatch` inside of the custom node.

**Reply by mueen-memon** [reply]
can you share sandbox? I am also trying to use redux.

**Reply by FredrikAugust** [reply]
Sadly, I don't have access to the code base anymore, but the clue lies in using redux for managing node-state, and passing around your own context.

**Reply by mueen-memon** [reply]
Thanks for the tip.

**Reply by moklick** [reply]
We will add a guide about how to use React Flow + Zustand (it's similar to Redux) soon
.

**Reply by mueen-memon** [reply]
Great. Please let me know once it is published. Thanks.
